### PR TITLE
PAY-1469

### PR DESCRIPTION
**Pull request by CrazyFunker**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PAY-1469

### Change description ###
Updated tests for the scenario. Removed FINREM as the only service that can use the functionality. Updated error messages

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java [hunk 1]
```diff
@@ -210,7 +215,14 @@ public String return504(AccountServiceUnavailableException ex) {
     }
 
     private boolean isAccountStatusCheckRequired(Service service) {
-        return ff4j.check("credit-account-payment-liberata-check") && Service.FINREM == service;
+        if (ff4j.check("credit-account-payment-liberata-check")) {
+            if (ff4j.check("check-liberata-account-for-all-services")) {
```
Comment (smathangi) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java]: what if only check-liberata-account-for-all-services is enabled?

File: model/src/main/java/uk/gov/hmcts/payment/api/service/FakeAccountServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import uk.gov.hmcts.payment.api.dto.AccountDto;
+import uk.gov.hmcts.payment.api.util.AccountStatus;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+@Profile("mock-liberata")
+@Primary
+@Service
+public class FakeAccountServiceImpl implements AccountService<AccountDto, String> {
```
Comment (smathangi) [inline model/src/main/java/uk/gov/hmcts/payment/api/service/FakeAccountServiceImpl.java]: you can use @MockBean in that controller test and define mock mappings there itself.

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/dsl/PaymentsTestDsl.java [hunk 3]
```diff
@@ -61,6 +61,13 @@ public PaymentGivenDsl returnUrl(String url) {
             return this;
         }
 
+        public PaymentGivenDsl setFF4Jfeature(String featureName, boolean value) {
+            String path = "/api/ff4j/store/features/" + featureName + (value ? "enable" : "disable");
```
Comment (smathangi) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/dsl/PaymentsTestDsl.java]: atm ff4j flags are stored in memory.. so there is no guarantee  which instance is updated (aat has 2 instances). So, the tests become flakey until we address flags persistency.

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java (hunk 1 above)
Comment (CrazyFunker) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java]: the test givenLiberataCheckOffAndCheckLiberataAccountForAllSericesOffThenNoServiceTriggersLiberataCheck() covers that case

File: model/src/main/java/uk/gov/hmcts/payment/api/service/FakeAccountServiceImpl.java (hunk 2 above)
Comment (CrazyFunker) [reply model/src/main/java/uk/gov/hmcts/payment/api/service/FakeAccountServiceImpl.java]: This is for developer-runtime only, not for tests so not used in tests

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/dsl/PaymentsTestDsl.java (hunk 3 above)
Comment (CrazyFunker) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/dsl/PaymentsTestDsl.java]: ok, for now I'm not using this functionality anywhere in the functional tests, but might come in handy later

**Pull request merged**